subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/JavaToolchainInput.java

```diff
@@ -32,6 +33,7 @@ public JavaToolchainInput(JavaToolchainSpec spec) {
         this.implementation = spec.getImplementation().get().toString();
     }
 
+    @Optional
```

Comment:
Shall we not add this as an input here and always use the one from the toolchain itself which is now an input? So the major version is always an input when you use toolchains as `@Nested`, whether you requested the version or not?

Reply:
That's a good point. If we exclude the language version from being an input in this class, I don't see much value in having the class itself as we can just make these values protected inputs on the `JavaToolchain` itself.

Regardless of which approach we choose, I suggest we implement it in the follow-ups.